controllers/helmrelease_controller.go

```diff
@@ -243,16 +243,19 @@ func (r *HelmReleaseReconciler) reconcileChart(ctx context.Context, hr *v2.HelmR
 	if err != nil && !apierrors.IsNotFound(err) {
 		return nil, false, err
 	}
+	hc := helmChartFromTemplate(*hr)
 	switch {
 	case apierrors.IsNotFound(err):
-		hc := helmChartFromTemplate(*hr)
 		if err = r.Client.Create(ctx, hc); err != nil {
 			return nil, false, err
 		}
 		hr.Status.HelmChart = chartName.String()
 		return nil, false, nil
 	case helmChartRequiresUpdate(*hr, helmChart):
-		hc := helmChartFromTemplate(*hr)
+		helmChart.Spec.Name = hc.Spec.Name
```

Comment:
I would do `helmChart.Spec = hc.Spec` so we don't need to update this code if we add more fields later on.